pkg/roll/run_update.go

```diff
@@ -167,13 +167,44 @@ ROLL_LOOP:
 				break ROLL_LOOP
 			}
 
-			next := algorithm(oldNodes.Healthy, newNodes.Healthy, u.DesiredReplicas, u.MinimumReplicas)
+			next := rollAlgorithm(oldNodes.Healthy, newNodes.Healthy, u.DesiredReplicas, u.MinimumReplicas)
 			if next > 0 {
+				// apply the delay only if we've already added to the new RC, since there's
+				// no value in sitting around doing nothing before anything has happened.
+				if newNodes.Desired > 0 && u.RollDelay > time.Duration(0) {
+					u.logger.WithField("delay", u.RollDelay).Infof("Waiting %v before continuing deploy", u.RollDelay)
```

Comment:
I don't mind having fully expanded human messages as well as metadata in the logs. It gives us the opportunity in the future to both display the message and also annotate it. For example, I imagine printing this message on our dashboard with a :clock10: adjacent to it, indicating that the system is waiting. It seems reasonable to want a field to indicate that icon should be displayed.

Is it the duplicativeness that bothers you?